src/components/ImportGeomFile/ImportGeomFileOverlay/ImportGeomFileOverlay.js

```diff
@@ -0,0 +1,186 @@
+import React, { useState, useEffect, useMemo } from 'react';
+import PropTypes from 'prop-types';
+import { Trans, useTranslation } from 'react-i18next';
+import { Button, Callout, FormGroup, InputGroup, Intent } from '@blueprintjs/core';
+import * as toGeojson from '@tmcw/togeojson';
+
+import {
+  ALL,
+  MULTI_POINT,
+  MULTI_LINESTRING,
+  MULTI_POLYGON,
+  geomTypeList,
+} from '../../../utils/geom';
+
+// '@tmcw/togeojson' package can only convert KML, TCX and GPX files for the moment
+const ALL_ACCEPTED_EXTENSIONS = Object.keys(toGeojson).filter(method => !method.endsWith('Gen'));
+
+const getGeojson = (data, fileExtension) => {
+  let geom = null;
+  let error = null;
+  const xml = new DOMParser().parseFromString(global.atob(data), 'text/xml');
+  try {
+    const xmlConverter = toGeojson[fileExtension];
+    geom = xmlConverter(xml);
+  } catch ({ name }) {
+    error = name;
+  }
+  return { geom, error };
+};
+
+const getFeatures = (geom, geomType, t) => {
+  if (!geom) {
+    return {
+      features: null,
+    };
+  }
+
+  const { features } = geom;
+  const geomTypeName = geomTypeList[geomType];
+  const isMultiGeom = [ALL, MULTI_POINT, MULTI_LINESTRING, MULTI_POLYGON].includes(geomType);
+  const filteredGeom = features.filter(({ geometry: { type } }) => type === geomTypeName);
+
+  const warnings = [];
+  if (features.length > filteredGeom.length) {
+    warnings.push(t('importGeomFile.warnGeomOtherType'));
+  }
+
+  if (isMultiGeom) {
+    return {
+      features: filteredGeom,
+      warnings,
+    };
+  }
+
+  if (filteredGeom.length > 1) {
+    warnings.push(t('importGeomFile.warnGeomSameType'));
+  }
+
+  const [firstItem] = filteredGeom;
+
+  if (!firstItem) {
+    warnings.push(t('importGeomFile.warnNoGeomFound'));
+  }
+
+  return {
+    features: firstItem ? [firstItem] : [],
+    warnings,
+  };
+};
+
+const ImportGeomFileOverlay = ({ geomType, onChange, acceptedExtensions: extensions }) => {
+  const [geojson, setGeojson] = useState({});
+  const [fileName, setFileName] = useState(null);
+
+  const { t } = useTranslation();
+
+  const acceptedExtensions = useMemo(() => (
+    extensions.filter(ext => ALL_ACCEPTED_EXTENSIONS.includes(ext))
+  ), [extensions]);
+
+
+  const handleReset = () => {
+    setFileName(null);
+    setGeojson({});
+  };
+
+  const handleChange = ({ target: { files: [file] } }) => {
+    handleReset();
+    const { name } = file;
+    setFileName(name);
+    const [, fileExtension] = name.split('.');
+    if (!acceptedExtensions.includes(fileExtension)) {
+      return;
+    }
+
+    const reader = new FileReader();
+    reader.onload = ({ target: { result } }) => {
+      const [, data] = result.split('base64,');
+      setGeojson(getGeojson(data, fileExtension));
+    };
+    reader.readAsDataURL(file);
```

Comment:
readAsText ? To avoid the split l.98 and the .atob l.21

Reply:
Top! Thx for the tip ❤️